WeakAuras/GenericTrigger.lua

```diff
@@ -1013,12 +1043,17 @@ function LoadEvent(id, triggernum, data)
       unit = string.lower(unit)
       for index, event in pairs(events) do
         MultiUnitLoop(
-          function(unit)
-            loaded_unit_events[unit] = loaded_unit_events[unit] or {};
-            loaded_unit_events[unit][event] = loaded_unit_events[unit][event] or {};
-            loaded_unit_events[unit][event][id] = loaded_unit_events[unit][event][id] or {}
-            loaded_unit_events[unit][event][id][triggernum] = data;
-          end, unit
+          function(u)
+            loaded_unit_events[u] = loaded_unit_events[u] or {};
+            loaded_unit_events[u][event] = loaded_unit_events[u][event] or {};
+            loaded_unit_events[u][event][id] = loaded_unit_events[u][event][id] or {}
+            loaded_unit_events[u][event][id][triggernum] = data;
+            if WeakAuras.UnitIsPet(u) then
```

Comment:
Is this correct?

Don't we need to add UNIT_PET for the pet owner units?

That is, I think the check here should be reversed?

Also is inside the loop the right place for the logic? Shouldn't adding UNIT_PET be the responsible of the caller?